lib/date_time/date_time.c

```diff
@@ -99,6 +113,59 @@ static void date_time_schedule_update(void)
 	}
 }
 
+#if defined(CONFIG_DATE_TIME_AUTO)
+static void date_time_store(int64_t curr_time_ms)
+{
+	struct timespec tp = { 0 };
+	int ret;
+
+	tp.tv_sec = curr_time_ms / 1000;
+	tp.tv_nsec = (curr_time_ms % 1000) * 1000000;
+
+	ret = clock_settime(CLOCK_REALTIME, &tp);
+	if (ret != 0) {
+		LOG_ERR("Could not set date %d", ret);
+		return;
+	}
+	LOG_DBG("System time updated");
+
+	/* Set modem time if it's not valid */
+	/* TODO: KCONFIG ON WHETHER MODEM TIME IS UPDATED? */
+	if (!modem_valid_time) {
+		/* TODO: We likely need to run time_modem_get here to confirm
+		 * there is no time so that we don't override it
+		 */
+		struct tm ltm = { 0 };
+
+		gmtime_r(&tp.tv_sec, &ltm);
+
+		/* AT+CCLK=<time>
+		 * "<time> sring type value; format is ""yy/MM/dd,hh:mm:ss±zz"",
+		 *         where characters indicate year (two last digits),
+		 *         month, day, hour, minutes, seconds and time zone
+		 *         (indicates the difference, expressed in quarters of an hour,
+		 *         between the local time and GMT; range -48...+48)."
+		 * Example command:
+		 *   "AT+CCLK="18/12/06,22:10:00+08"
+		 */
+
+		/* Time zone is not known and it's mandatory so setting to zero.
+		 * POSIX year is relative to 1900 which doesn't affect as last two digits are taken
+		 * with modulo 100.
+		 * POSIX month is in range 0-11 so adding 1.
+		 */
+		ret = nrf_modem_at_printf("AT+CCLK=\"%02u/%02u/%02u,%02u:%02u:%02u+%02u\"",
```

Comment:
It would be nice to have the date and time debug-logged.  Perhaps render the portion after the = to a local buffer, LOG_DBG() it, and then use it to format the string for `nrf_modem_at_printf()`.